libnopegl/src/pass.c

```diff
@@ -174,21 +169,32 @@ static int register_texture(struct pass *s, const char *name, struct ngl_node *t
                 texture_priv->params.usage |= NGLI_TEXTURE_USAGE_STORAGE_BIT;
 
                 if (texture->cls->id == NGL_NODE_TEXTURE2D)
-                    crafter_texture.type = NGLI_PGCRAFT_SHADER_TEX_TYPE_IMAGE_2D;
+                    type = NGLI_PGCRAFT_SHADER_TEX_TYPE_IMAGE_2D;
                 else if (texture->cls->id == NGL_NODE_TEXTURE2DARRAY)
-                    crafter_texture.type = NGLI_PGCRAFT_SHADER_TEX_TYPE_IMAGE_2D_ARRAY;
+                    type = NGLI_PGCRAFT_SHADER_TEX_TYPE_IMAGE_2D_ARRAY;
                 else if (texture->cls->id == NGL_NODE_TEXTURE3D)
-                    crafter_texture.type = NGLI_PGCRAFT_SHADER_TEX_TYPE_IMAGE_3D;
+                    type = NGLI_PGCRAFT_SHADER_TEX_TYPE_IMAGE_3D;
                 else if (texture->cls->id == NGL_NODE_TEXTURECUBE)
-                    crafter_texture.type = NGLI_PGCRAFT_SHADER_TEX_TYPE_IMAGE_CUBE;
+                    type = NGLI_PGCRAFT_SHADER_TEX_TYPE_IMAGE_CUBE;
```

Comment:
Already the case in the existing code but why is this almost exactly the same as the `type` assignment at the beginning of the function (except one is a switch case and the other is a bunch of if)?

Reply:
One is for texture/sampler types, the other one is for image (as in image load/store) types. I have some commit on the way that extract this to dedicated function. it makes the code cleaner (and the same style is used for both case).